### PR TITLE
Fix integer overflow in dump for split thresholds beyond int32 range

### LINKED ISSUE #10035: Integer overflow in `get_dump` and `trees_to_dataframe`
Dumping tree information using `get_dump` and `trees_to_dataframe` in Python for models with large split conditions causes integer overflow errors.
This has only been tested with Python regression models on v2.0.3.

### How to reproduce
In python:

```
model = XGBRegressor(n_jobs=100, n_estimators=1, min_child_weight=1, subsample=1, learning_rate=0.05, max_depth=4, random_state=1)
input = pd.DataFrame({'very_large_number': [2*10**10, 3*10**10]})  
output = pd.DataFrame({'true_results': [0, 1]})

model.fit(input, output)

model.get_booster().trees_to_dataframe()
>>>    Tree  Node   ID            Feature         Split  Yes   No Missing    Gain  Cover  Category
>>> 0     0     0  0-0  very_large_number -2.147484e+09  0-1  0-2     0-2  0.2500    2.0       NaN
>>> 1     0     1  0-1               Leaf           NaN  NaN  NaN     NaN -0.0125    1.0       NaN
>>> 2     0     2  0-2               Leaf           NaN  NaN  NaN     NaN  0.0125    1.0       NaN

model.predict([-10, 2*10**10, 3*10**10])
>>> [0.4875 0.4875 0.5125]
```

Note the split condition for feature `very_large_number`: `-2.147484e+09 == -INT_MAX`.

If we are to trust the output of `trees_to_dataframe`, each prediction for `[-10, 2*10**10, 3*10**10]` should return `[0.5125, 0.5125, 0.5125]` since each of those values are larger than `-2.147484e+09`. However, this is not what is observed from the model. 

The same integer overflow error can be observed when using `model.get_booster().get_dump()`.

However, when using `model.save_model(file)` the split conditions are saved correctly. This can be tested by saving and loading, in which case you will observe identical model outputs indicating there is no data loss in the process.

The `trees_to_dataframe()` function calls the underlying `get_dump()` which in turn accesses `XGBoosterDumpModelEx`. However, `save_model()` uses `XGBoosterSaveModel`. Thus, it seems like the problem is likely with `XGBoosterDumpModelEx`.

This is my first time contributing to this project (or any open source project). Please let me know if there are any issues or if any more information is needed. 
Thanks!

### PR DESCRIPTION
**Pull request by momomuchu**
Fixes #10035.

`get_dump()` and `trees_to_dataframe()` report a garbage threshold (INT32_MIN or INT32_MAX) instead of the real split condition when a `kInteger` feature's threshold exceeds the int32 range. For example a model split at ~3e10 dumps `2147483647`. `predict()` is unaffected: the predictor compares against the raw float `SplitCond` and never goes through the dump formatter, so only the text/JSON/dataframe dumps were wrong.

Root cause: the two `Integer()` dump formatters in `src/tree/tree_model.cc` (`TextGenerator` and `JsonGenerator`) do an unchecked `static_cast<int32_t>` on the floored split condition, which wraps around (and is UB for out-of-range values).

This widens the cast to `int64_t` so the real threshold is reported, and clamps to the int64 range before the cast so a value beyond int64 (only reachable if a >9.2e18 float feature is mislabeled as int) saturates deterministically rather than being UB. The clamp is a no-op for every value a genuine int feature can hold. Both dump paths and `trees_to_dataframe` now report the true threshold.

Added tests for the int64 overflow, the in-range boundary (no regression), float thresholds being unaffected, and the beyond-int64 saturation. `plot_tree`/graphviz was checked and never had this bug (it renders the raw float, not the integer formatter), so it is left unchanged.

One open question for you: I widened to int64 to preserve the value; the alternative is clamping to int32 with a warning if you would rather keep a strict 32-bit dump range for downstream consumers. Happy to switch if you prefer.

### CODE REVIEW

**Comment by trivialfis** [comment]
I think we can fix it by simply using `int64_t` or `uint64_t` since we don't support larger integers for training anyway.
